cpp/src/arrow/util/bit_util_test.cc

```diff
@@ -315,6 +318,115 @@ TEST(FirstTimeBitmapWriter, NormalOperation) {
   }
 }
 
+std::string BitmapToString(const uint8_t* bitmap, int64_t bit_count) {
+  return arrow::internal::Bitmap(bitmap, /*offset*/ 0, /*length=*/bit_count).ToString();
+}
+
+std::string BitmapToString(const std::vector<uint8_t>& bitmap, int64_t bit_count) {
+  return BitmapToString(bitmap.data(), bit_count);
+}
+
+TEST(FirstTimeBitmapWriter, AppendWordOffsetOverwritesCorrectBitsOnExistingByte) {
+  auto check_append = [](const std::string& expected_bits, int64_t offset) {
+    std::vector<uint8_t> valid_bits = {0x00};
```

Comment:
For each of the tests here it would be nice to have two variants or iterations: one that starts with a bitmap full of 0s, and one with a bitmap full of 1s.

Reply:
I added variants to this test.  Note that I don't think have a bitmap full of 111s iand overwriting trailing ones isn't really expected because any not processed are expected to be zero.

Reply:
Well, `FirstTimeBitmapWriter` is supposed to work on uninitialized data.